### PR TITLE
Worked on issue #361: Updated string API in notebooks

### PR DESCRIPTION
**Pull request by arp95**
Worked on issue #361 .@Nic-Ma Please check.

### CODE REVIEW

File: examples/notebooks/cache_dataset_speed.ipynb [hunk 1]
```diff
@@ -180,11 +180,10 @@
     "            loss.backward()\n",
     "            optimizer.step()\n",
     "            epoch_loss += loss.item()\n",
-    "            print('{}/{}, train_loss: {:.4f}, step time: {:.4f}'.format(\n",
-    "                  step, len(train_ds) // train_loader.batch_size, loss.item(), time.time() - step_start))\n",
+    "            print(f\"{step}/{len(train_ds) // train_loader.batch_size}, train_loss: {loss.item()}, step time: {time.time() - step_start}\")\n",
     "        epoch_loss /= step\n",
     "        epoch_loss_values.append(epoch_loss)\n",
-    "        print('epoch {} average loss: {:.4f}'.format(epoch + 1, epoch_loss))\n",
+    "        print(f\"epoch {epoch + 1} average loss: {epoch_loss}\")\n",
```
Comment (Nic-Ma) [inline examples/notebooks/cache_dataset_speed.ipynb:186]: The previous string only keeps 4 places after the decimal point.
Something like: `f"XXX{data: 4}"`

Thanks.

File: examples/notebooks/cache_dataset_speed.ipynb [hunk 2]
```diff
@@ -180,11 +180,10 @@
     "            loss.backward()\n",
     "            optimizer.step()\n",
     "            epoch_loss += loss.item()\n",
-    "            print('{}/{}, train_loss: {:.4f}, step time: {:.4f}'.format(\n",
-    "                  step, len(train_ds) // train_loader.batch_size, loss.item(), time.time() - step_start))\n",
+    "            print(f\"{step}/{len(train_ds) // train_loader.batch_size}, train_loss: {loss.item()}, step time: {time.time() - step_start}\")\n",
```
Comment (Nic-Ma) [inline examples/notebooks/cache_dataset_speed.ipynb:183]: This line seems too long than 120 chars.

File: examples/notebooks/cache_dataset_speed.ipynb [hunk 3]
```diff
@@ -207,12 +206,10 @@
     "                    best_metric_epoch = epoch + 1\n",
     "                    torch.save(model.state_dict(), 'best_metric_model.pth')\n",
     "                    print('saved new best metric model')\n",
-    "                print('current epoch: {} current mean dice: {:.4f} best mean dice: {:.4f} at epoch {}'.format(\n",
-    "                    epoch + 1, metric, best_metric, best_metric_epoch))\n",
-    "        print('time consuming of epoch {} is: {:.4f}'.format(epoch + 1, time.time() - epoch_start))\n",
+    "                print(f\"current epoch: {epoch + 1} current mean dice: {metric} best mean dice: {best_metric} at epoch {best_metric_epoch}\")\n",
```
Comment (Nic-Ma) [inline examples/notebooks/cache_dataset_speed.ipynb:209]: The previous string only keeps 4 places after the decimal point.
Something like: `f"XXX{data: 4}"`

Thanks.

File: examples/notebooks/mednist_tutorial.ipynb [hunk 4]
```diff
@@ -398,9 +398,8 @@
     "                best_metric_epoch = epoch + 1\n",
     "                torch.save(model.state_dict(), 'best_metric_model.pth')\n",
     "                print('saved new best metric model')\n",
-    "            print('current epoch: {} current AUC: {:.4f} current accuracy: {:.4f} best AUC: {:.4f} at epoch {}'.format(\n",
-    "                epoch + 1, auc_metric, acc_metric, best_metric, best_metric_epoch))\n",
-    "print('train completed, best_metric: {:.4f} at epoch: {}'.format(best_metric, best_metric_epoch))"
+    "            print(f\"current epoch: {epoch + 1} current AUC: {auc_metric} current accuracy: {acc_metric} best AUC: {best_metric} at epoch {best_metric_epoch}\")\n",
```
Comment (Nic-Ma) [inline examples/notebooks/mednist_tutorial.ipynb:401]: This line seems too long than 120 chars.

File: examples/notebooks/persistent_dataset_speed.ipynb [hunk 5]
```diff
@@ -112,11 +112,10 @@
     "            loss.backward()\n",
     "            optimizer.step()\n",
     "            epoch_loss += loss.item()\n",
-    "            print('{}/{}, train_loss: {:.4f}, step time: {:.4f}'.format(\n",
-    "                  step, len(train_ds) // train_loader.batch_size, loss.item(), time.time() - step_start))\n",
+    "            print(f\"{step}/{len(train_ds) // train_loader.batch_size}, train_loss: {loss.item()}, step time: {time.time() - step_start}\")\n",
     "        epoch_loss /= step\n",
     "        epoch_loss_values.append(epoch_loss)\n",
-    "        print('epoch {} average loss: {:.4f}'.format(epoch + 1, epoch_loss))\n",
+    "        print(f\"epoch {epoch + 1} average loss: {epoch_loss}\")\n",
```
Comment (Nic-Ma) [inline examples/notebooks/persistent_dataset_speed.ipynb:118]: The previous string only keeps 4 places after the decimal point.
Something like: `f"XXX{data: 4}"`

Thanks.

File: examples/notebooks/persistent_dataset_speed.ipynb [hunk 6]
```diff
@@ -139,12 +138,10 @@
     "                    best_metric_epoch = epoch + 1\n",
     "                    torch.save(model.state_dict(), 'best_metric_model.pth')\n",
     "                    print('saved new best metric model')\n",
-    "                print('current epoch: {} current mean dice: {:.4f} best mean dice: {:.4f} at epoch {}'.format(\n",
-    "                    epoch + 1, metric, best_metric, best_metric_epoch))\n",
-    "        print('time consuming of epoch {} is: {:.4f}'.format(epoch + 1, time.time() - epoch_start))\n",
+    "                print(f\"current epoch: {epoch + 1} current mean dice: {metric} best mean dice: {best_metric} at epoch {best_metric_epoch}\")\n",
```
Comment (Nic-Ma) [inline examples/notebooks/persistent_dataset_speed.ipynb:141]: The previous string only keeps 4 places after the decimal point.
Something like: `f"XXX{data: 4}"`

Thanks.

File: examples/notebooks/persistent_dataset_speed.ipynb [hunk 7]
```diff
@@ -311,7 +307,7 @@
     "\n",
     "cache_epoch_num, cache_total_time, cache_epoch_loss_values, cache_metric_values, cache_epoch_times = \\\n",
     "    train_process(cache_train_ds, cache_val_ds)\n",
-    "print('Total training time of {} epochs with CacheDataset: {:.4f}'.format(cache_epoch_num, cache_total_time))"
+    "print(f\"Total training time of {cache_epoch_num} epochs with CacheDataset: {cache_total_time}\")"
```
Comment (Nic-Ma) [inline examples/notebooks/persistent_dataset_speed.ipynb:310]: The previous string only keeps 4 places after the decimal point.
Something like: `f"XXX{data: 4}"`

Thanks.

File: examples/notebooks/spleen_segmentation_3d.ipynb [hunk 8]
```diff
@@ -311,10 +311,10 @@
     "        loss.backward()\n",
     "        optimizer.step()\n",
     "        epoch_loss += loss.item()\n",
-    "        print('{}/{}, train_loss: {:.4f}'.format(step, len(train_ds) // train_loader.batch_size, loss.item()))\n",
+    "        print(f\"{step}/{len(train_ds) // train_loader.batch_size}, train_loss: {loss.item()}\")\n",
```
Comment (Nic-Ma) [inline examples/notebooks/spleen_segmentation_3d.ipynb:314]: The previous string only keeps 4 places after the decimal point.
Something like: `f"XXX{data: 4}"`

Thanks.

**Comment by arp95** [comment]
@Nic-Ma  Updated. Please check.

**Pull request merged**